crates/librqbit/src/torrent_state/live/peer/mod.rs

```diff
@@ -16,10 +18,19 @@ pub(crate) type InflightRequest = ChunkInfo;
 pub(crate) type PeerRx = UnboundedReceiver<WriterRequest>;
 pub(crate) type PeerTx = UnboundedSender<WriterRequest>;
 
+#[derive(Debug, Default)]
+pub(crate) enum OutgoingAddressType {
```

Comment:
Actually I started with Option,  but later I thought that I would need to have 3 states, Default, if peers key is outgoing address, None if we don't know outgoing address for incoming peer and Known, if we know outgoing address for incoming peer.   
But on second thought  maybe Option will be better, in that case, now will need to duplicate their SocketAddress / key for outgoing peers,   but it may be more future proof,  if peers map will not use SocketAddress as key. Relying on key special semantic is probably not ideal anyhow.

Will look to it and change it.